deployment/common/Configuration.psm1

```diff
@@ -612,6 +612,12 @@ function Get-SreConfig {
     # Import minimal management config parameters from JSON config file - we can derive the rest from these
     $sreConfigBase = Get-CoreConfig -shmId $shmId -sreId $sreId
 
+    # "MicrosoftRDS" is deprecated and the "remotedDesktopProvider" field now defaults to "ApacheGuacamole"
+    if ($sreConfigBase.remoteDesktopProvider -eq "MicrosoftRDS") {
+        Add-LogMessage -Level Warning "MicrosoftRDS support is deprecated. Apache Guacamole will be used to provide Remote Desktop services."
+    }
+    $sreConfigBase.remoteDesktopProvider = "ApacheGuacamole"
```

Comment:
I don't think this does what we want. A user could put in any invalid except "MicrosoftRDS" and would get no warning or error. A user can explicitly select "MicrosoftRDS" and the process will continue, deploying something else instead.

```suggestion
    if ($sreConfigBase.remoteDesktopProvider -ne "ApacheGuacamole") {
        Add-LogMessage -Level Fatal "Support for remote desktops other than ApacheGuacamole has been removed"
    } elseif ($sreConfigBase.remoteDesktopProvider -eq "ApacheGuacamole") {
        Add-LogMessage -Level Warning "The remoteDesktopProvider configuration option has been deprecated and will be removed in the future"
    }
    $sreConfigBase.remoteDesktopProvider = "ApacheGuacamole"
```

or, would this work

```powershell
if ($sreConfigBase.remoteDesktopProvider -ne $null) {
        Add-LogMessage -Level Fatal "Support for remote desktops other than ApacheGuacamole has been removed. Remove the remoteDesktopProvider option from your configuration."
}
$sreConfigBase.remoteDesktopProvider = "ApacheGuacamole"
```